watertap/unit_models/crystallizer.py

```diff
@@ -316,6 +329,20 @@ def eq_mass_balance_constraints(b, j):
                 if (p, j) in b.properties_solids[0].phase_component_set
             )
 
+        @self.Constraint()
+        def eq_pure_vapor_flow_rate(b):
+            return (
+                b.properties_pure_water[0].flow_mass_phase_comp["Vap", "H2O"]
+                == b.properties_vapor[0].flow_mass_phase_comp["Vap", "H2O"]
+            )
+
+        self.properties_pure_water[0].flow_mass_phase_comp["Liq", "H2O"].fix(1e-8)
+        self.properties_pure_water[0].flow_mass_phase_comp["Liq", "NaCl"].fix(0)
+        self.properties_pure_water[0].flow_mass_phase_comp["Sol", "NaCl"].fix(0)
+        self.properties_pure_water[0].mass_frac_phase_comp["Liq", "NaCl"]
+        self.properties_in[0].conc_mass_phase_comp["Liq", "NaCl"]
+        self.properties_in[0].flow_vol_phase["Liq"]
```

Comment:
Why touch these properties here and now within the unit model? My guess is that you want them to be calculated when reporting results after a solve (so a matter of convenience), but that could technically be done at the flowsheet level without introducing within the unit.

Reply:
Yea I realized that after I pushed the PR. Thanks! Will clean it up in the next commit.